Merging develop into master

mexanick opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
Code refactoring, fixing all the issue w.r.t. indentation and dilepton collections
## Description
<!--- Describe your changes in detail -->

### Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue(s) here. -->
Fixes all the indentation problems, including the one leading to the ProductNotFound error

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
Tested with CMSSW_8_0_29 by running locally with cmsRun

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

<!--- Template thanks to https://www.talater.com/open-source-templates/#/page/99 -->

mexanick committed:
Merge pull request #3 from mexanick/develop

Fixing CONTRIBUTING.md

rathjd committed:
Merge pull request #5 from mexanick/develop

Cleaning up excessive binaries
Improving python tabbing > spacing
Extending jet collections in ntuples from 4 to 10

mexanick committed:
Resructuring folders

mexanick committed:
Fixing indentation and couple of other minor bugs

mexanick committed:
Removing unnecessary file paths

mexanick committed:
Updating README

mexanick committed:
Update README

mexanick committed:
Update README.md

mexanick committed:
Update README.md

mexanick committed:
Update histosSimplified.py

* Reverting default flags settings and number of events to process
* Adding dielectron collections

rathjd committed:
Merge pull request #8 from mexanick/refactoring_develop

Dielectron selection -> needs to be tested
Code improvements

mexanick committed:
Fixing indentation error leading to the ProductNotFound error

rathjd commented on test/DataMCSpectraComparison/histosSimplified.py:
```diff
@@ -13,9 +13,9 @@
+'/store/mc/RunIISummer16MiniAODv2/ZToMuMu_NNPDF30_13TeV-powheg_M_120_200/MINIAODSIM/PUMoriond17_80X_mcRun2_asymptotic_2016_TrancheIV_v6-v1/80000/824C363B-0AC8-E611-B4A5-20CF3027A580.root',
 			   ]
-process.maxEvents.input = -1
+process.maxEvents.input = 100 # Set to a reasonable number (e.g.100) when testing locally with cmsRun
```
Back to 100 in the main code? I'd advise setting this back to -1, or at least remember doing that when it propagates from the development to the main branch.

mexanick replied:
oups... missed it again... will fix

rathjd commented on test/DataMCSpectraComparison/histosSimplified.py:
```diff
@@ -278,21 +283,34 @@ def ntuplify(process, fill_gen_info=False):
                                            vertices_src = cms.InputTag('offlineSlimmedPrimaryVertices'),
-                                           #TriggerResults_src = cms.InputTag('TriggerResults', '', 'PAT'),	#mc
-                                           TriggerResults_src = cms.InputTag('TriggerResults', '', 'RECO'),	#data
+                                           TriggerResults_src = cms.InputTag('TriggerResults', '', 'PAT'),	#mc
```
Can we do this more generically with an isData switch?

mexanick replied:
fair point

rathjd reviewed (commented):
Thanks for making the setup instructions clearer and fixing the commentation markers from C++ to python logic.
There are still some general improvements that ought to be made, including an automatic switch between MC and data reconstruction.

hyeahyun commented on README.md:
````diff
@@ -1,48 +1,45 @@
 
+```bash
+cmsrel CMSSW_8_0_29
 cd CMSSW_8_0_25/src
````
There is a small typo. 
cd CMSSW_8_0_25/src  ->  cd CMSSW_8_0_29/src

hyeahyun commented on README.md:
```diff
@@ -1,48 +1,45 @@
 cmsenv
-
+mkdir SUSYBSMAnalysis && cd SUSYBSMAnalysis/
 git clone git@github.com:BFF-Zprime-mumu/Zprime2muAnalysis.git
```
Must complete the electron HEEP ID setup before downloading our analysis codes.
If it does not, you get an error corresponding to the # 1 error on the slide that I announced this week.

hyeahyun reviewed (changes requested):
I think there is nothing wrong with the codes overall.

hyeahyun reviewed (changes requested):
I think there is nothing wrong with the code overall.
I only request changes for minor things in the setup process.

mexanick committed:
Update README.md

mexanick committed:
Update README.md

mexanick committed:
Update README.md

mexanick committed:
Update histosSimplified.py

Fixing the nevents and automated trigger results input tag name (default for data, switched for MC on line 318)

mexanick commented:
All comments has been addressed, please check @rathjd @hyeahyun

hyeahyun reviewed (approved):
Thank you for your quick feedback.
I will push the codes I fixed as soon as possible.

rathjd reviewed (approved):
Everything is in order now, it seems.
